src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -159,13 +222,42 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                     },
                     Preselect = item.Rules.SelectionBehavior == CompletionItemSelectionBehavior.HardSelection,
                 };
+
                 var commitCharacters = GetCommitCharacters(item, commitCharacterRulesCache);
                 if (commitCharacters != null)
                 {
                     completionItem.CommitCharacters = commitCharacters;
                 }
 
                 return completionItem;
+
+                static async Task<LSP.TextEdit> GenerateTextEdit(
+                    Document document,
+                    CompletionItem item,
+                    CompletionService completionService,
+                    SourceText? documentText,
+                    TextSpan? defaultSpan,
+                    LSP.Range? defaultRange,
+                    LSP.TextEdit? textEdit,
```

Comment:
i don't understand why this is passed in.

Reply:
Ah good point, thanks for the catch!